modules/sql-engine/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -787,3 +787,39 @@ Boolean SqlKillNoWait():
         return noWait;
     }
 }
+
+/**
+ * Parses an EXPLAIN PLAN statement.
+ */
+SqlNode SqlIgniteExplain() :
+{
+    SqlNode stmt;
+    SqlExplainLevel detailLevel = SqlExplainLevel.EXPPLAN_ATTRIBUTES;
+    SqlExplain.Depth depth;
+    Span s;
+    final SqlExplainFormat format;
+}
+{
+    <EXPLAIN> { s = span(); } <PLAN>
+    [ detailLevel = ExplainDetailLevel() ]
+    depth = ExplainDepth()
+    (
+        LOOKAHEAD(2)
+        <AS> <XML> { format = SqlExplainFormat.XML; }
+    |
+        LOOKAHEAD(2)
+        <AS> <JSON> { format = SqlExplainFormat.JSON; }
+    |
+        <AS> <DOT_FORMAT> { format = SqlExplainFormat.DOT; }
```

Comment:
```suggestion
        LOOKAHEAD(2)
        <AS> <DOT_FORMAT> { format = SqlExplainFormat.DOT; }
```

Reply:
Should we use LOOKAHEAD(2) here as well or it is enough to set it once for the very first option?

Reply:
It's enough to use it for first two options.

LOOKAHEAD is used when you have several options with the same prefix. For instance, without LA(2) parser will choose `<AS> <XML>` as soon as it see `<AS>` token. With LA(2) it check the next token as well. Given that there is nothing to choose from after `<AS> <DOT_FORMAT>` it is OK to omit LA(2)